disnake/ext/commands/params.py

```diff
@@ -180,18 +182,86 @@ def _xt_to_xe(xe: Optional[float], xt: Optional[float], direction: float = 1) ->
 
 
 class Injection:
+    """Represents a slash command injection
+
+    .. versionadded:: 2.3
+
+    .. versionchanged:: 2.6
+        Constructor now accept keyword-only argument ``autocompleters``
+
+    Attributes
+    ----------
+    function: Callable
+        The underlying injection
+    """
+
     _registered: ClassVar[Dict[Any, Injection]] = {}
 
     function: Callable
+    autocompleters: Dict[str, Callable]
+
+    def __init__(self, function: Callable, *, autocompleters: Dict[str, Callable] = ...) -> None:
+        if not asyncio.iscoroutinefunction(function):
+            raise TypeError("Injection must be a function")
+
+        if autocompleters is not Ellipsis and any(
+            [not asyncio.iscoroutinefunction(f) for f in autocompleters.values()]
+        ):
+            raise TypeError("Some of your autocompleters are not coroutines")
 
-    def __init__(self, function: Callable) -> None:
         self.function = function
+        self.autocompleters = autocompleters if autocompleters is not Ellipsis else {}
 
     @classmethod
-    def register(cls, function: CallableT, annotation: Any) -> CallableT:
+    def register(
+        cls, function: Callable, annotation: Any, *, autocompleters: Dict[str, Callable] = ...
+    ) -> Self:
+        if not asyncio.iscoroutinefunction(function):
+            raise TypeError("Function must be a coroutine")
+
+        if autocompleters is not Ellipsis and any(
+            [not asyncio.iscoroutinefunction(f) for f in autocompleters.values()]
+        ):
+            raise TypeError("Some of your autocompleters are not coroutines")
+
         self = cls(function)
         cls._registered[annotation] = self
-        return function
+        self.autocompleters = autocompleters if autocompleters is not Ellipsis else {}
+        return self
+
+    def autocomplete(self, param_name: str) -> Callable:
+        """A decorator that adds an autocomplete for a given parameter
+
+        Parameters
+        ----------
+        param_name: :class:`str`
+            The name of the parameter to assign autocomplete to
```

Comment:
```suggestion
        """A decorator that registers an autocomplete function for the specified option.

        Parameters
        ----------
        param_name: :class:`str`
            The name of the option.
```
Essentially a copy of `InvokableSlashCommand.autocomplete`, for consistency